app/actions/views/channel.js

```diff
@@ -373,6 +374,21 @@ export function handleSelectChannel(channelId) {
     };
 }
 
+export function handleSelectChannelByName(channelName, teamName) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {teams: currentTeams, currentTeamId} = state.entities.teams;
+        const currentTeamName = currentTeams[currentTeamId].name;
+        const {data: channel} = await getChannelByNameAndTeamName(teamName || currentTeamName, channelName)(dispatch, getState);
```

Comment:
Instead of calling the result of `getChannelByNameAndTeamName`, it's actually easier to just dispatch it since that removes the explicit `getState` parameter and is usable for both async and regular actions.

```suggestion
        const {data: channel} = await dispatch(getChannelByNameAndTeamName(teamName || currentTeamName, channelName));
```